### ISSUE TITLE
Feature Request - Cockpit Tablet

### ISSUE THREAD

**Issue by AAL5665** [legoboyvdlp/A320-family#224]
Will it be possible to add a screen touch tablet(in relation to A350 MCDU's screen touch interface), if so, will it be possible to make it capable of doing the following:
1. Display simbrief related option such as:
i. Route(while showing the flight progress as well) 
ii. Time until arrival
iii. Callsign(Not necessary but a suggestion)
iv. Able to import simbrief flightplan OFP or printout(this might be asking for a lot)
v. Distance till you reach the Arrival Airport.

2. Control aircraft doors,
3. Connect External Power and Air.

**Comment by merspieler** [comment]
We have ideas for it... but won't come until near completion of the actual systems.

**Comment by AAL5665** [comment]
Sure fine by me

**Comment by legoboyvdlp** [comment]
> Control aircraft doors,

> Connect External Power and Air.

This doesn't make sense. Those are performed by ground crew; not in a "tablet" device.

**Comment by AAL5665** [comment]
Okay understood was referring to the FBW A320 EFB

**Comment by NewoIsTaken** [comment]
@legoboyvdlp X-Plane has a plugin called Avi-Tab and allows you to control those things in it. It is your choice whether you want to combine the two or keep them separate.

**Comment by Octal450** [comment]
Yes! I have been working on this on and off for the past few years... its something I will likely look into once the FDM (which is keeping most of my time rn) is complete!

Kind Regards,
Josh

**Comment by merspieler** [comment]
Oh... and just FYI @AAL5665  :
> ii. Time until arrival  

That will be visible on the MCDU, once completed  

> iii. Callsign(Not necessary but a suggestion)  

Already on the MCDU F-PLN page  

> iv. Able to import simbrief flightplan OFP or printout(this might be asking for a lot)  

You can already import from simbrief....  

> v. Distance till you reach the Arrival Airport.  

Again, already available on the F-PLN page... as well you can use the PROG pages BRG/DIST function to get the directe distance

**Comment by NewoIsTaken** [comment]
I was thinking instead of having menus to do stuff like push back or enable ground power, we put it in this tablet.

**Comment by merspieler** [comment]
doesn't make any sense... on a real tablet you wouldn't have that...

**Comment by NewoIsTaken** [comment]
But in real life, you wouldn't have menus either. Technically both are unrealistic. And this tablet isn't like an iPad tablet. It is a special tablet. :)

**Comment by merspieler** [comment]
well... at least we can keep the tablet realistic... by not including that.
... and some airlines actually use iPads...

**Comment by NewoIsTaken** [comment]
I believe that the idea for this tablet is not like a regular tablet. It is supposed to be like a control center for the plane but also a place for references like charts. Ultimately, this is you people's decision but those were just my two cents.

**Comment by merspieler** [comment]
Don't steal the bad ideas from the commercial sims :P

**Comment by Octal450** [comment]
I actually think its a good idea @merspieler (to put the acconfig menus and sh*t in there) :D 

Kind Regards,
Josh

**Comment by merspieler** [comment]
cause the commercial crap does it?

**Comment by inuyaksa** [comment]
My two cents. I don't like a fake tablet on this plane, I prefer a realistic table with all the functions a real pilot can use.
Other needed functions can be implemented on menu or acconfig.

**Comment by NewoIsTaken** [comment]
Hey, maybe add a toggle, if you like the idea, then enable it, if you don't, you can disable it and just use the menus! :)

**Comment by Octal450** [comment]
> cause the commercial crap does it?

Nah, just because its a convenient place to put stuff imo vs clumsy menus.

Not a requirement I just thought it would be cool. We can decide on it later anyways, take a vote and see who likes what.

Kind Regards,
Josh

**Comment by AAL5665** [comment]
Wow this is quite the discussion

**Comment by AAL5665** [comment]
> My two cents. I don't like a fake tablet on this plane, I prefer a realistic table with all the functions a real pilot can use.
> Other needed functions can be implemented on menu or acconfig.

Currently you are aware that the NAV mode only displays the F-PLN in a certain distance and the function of the route manager is none existing. Even if you use the F-PLN mode to view the result aren't different.

**Comment by AAL5665** [comment]
If there is a wish for such realism then there should be voice commands which might be used in controlling the ground crew(which might not be necessary) through the use of the MECH button for communication(initiation of commands).
Do not forget that in reality the doors are opened by the cabin crew which if you would want that then you might need that voice command option via the CALL button on the overhead panel to initiate the command.


Ignore If against the idea do not read the following:

Just know this is personal preference(suggestion) ignore if you must. 

Toggle in aircraft config to enable or disable VC option
Function: 
if mech_button press then VC(voice command) enable
then if mech_button press again disable VC

if call_button press then VC(voice command) enable
then if call_button press disable VC

Execution:
Mech commands:
Connect External Power
Connect External Air
Connect Catering Truck
Connect Baggage Ramp/Cart

Call commands:
Open Doors
Close Doors
Announcement(condition to be met before it can starts)

Just know this is personal preference(suggestion) ignore if you must.

**Comment by merspieler** [comment]
Here is your distance:
![fgfs-20210404231814](https://user-images.githubusercontent.com/29063255/113524137-344f9680-959c-11eb-8372-bdcc811b30d9.png)

There is already voice control for a few functions... https://github.com/merspieler/flight-gear-copilot tho this is a very generic implementation and hasn't been updated in a while.
Adding stuff to it would be easy tho

Actually that CABIN/MECH buttons aren't that much of a bad idea...
Voice commands are probably a bit overkill but if you click the MECH button and you get a dialog for the things you mentioned, that would be neat... maybe with spoken dialog....

**Comment by AAL5665** [comment]
In addition, could the PAX, Cargo, FOB be displayed. For example, 

if: 
Forward Cabin = 3722
Centre Cabin = 2721
Aft Cabin = 4233
then
PAX = Forward Cabin + Centre Cabin + Aft / 100

if total == 10676 / 100 then
total = 106.7 or 107

if:
Forward Cargo = 3000
Aft Cargo = 3000
then

Cargo = Forward Cargo + Aft Cargo / 1000
if Cargo == 6000/1000
Cargo = 6.0

Display on tablet in a tab(Fuel And Payload tab)
PAX: 106.7
Cargo: 6.0

**Comment by AAL5665** [comment]
By the way going of topic, I would like to make a PACX related mod for FlightGear but I am new to this and was wondering if you could give help on how to learn and get started?

**Comment by merspieler** [comment]
If you already know, you're off topic, then why do you post it in the first place???? (post removed)
Your question is better asked on the forum so I won't answer here-

For weight an pax... there was the idea of having an actual load sheet....

As said before, I'd prefer to have something similar to a real tablet on it and anything which is limited to the sim in a separate dialog.

Oh... and please DO NOT reference other sims, DO reference official documents or other references of the real thing.

**Comment by legoboyvdlp** [comment]
Real: 
![image](https://user-images.githubusercontent.com/10434911/114020521-97288280-9867-11eb-9594-9268218b8e31.png)
![image](https://user-images.githubusercontent.com/10434911/114020553-a0b1ea80-9867-11eb-860a-080159543215.png)
![image](https://user-images.githubusercontent.com/10434911/114020604-b0c9ca00-9867-11eb-9773-f08edb5faf24.png)
![image](https://user-images.githubusercontent.com/10434911/114020687-c2ab6d00-9867-11eb-936d-55c00019ea60.png)
![image](https://user-images.githubusercontent.com/10434911/114020768-d951c400-9867-11eb-9f6b-a429d8e0b971.png)
![image](https://user-images.githubusercontent.com/10434911/114020807-e1a9ff00-9867-11eb-96da-d427878ee0e9.png)
![image](https://user-images.githubusercontent.com/10434911/114020839-e8d10d00-9867-11eb-9886-ca9344c642f9.png)

**Comment by NewoIsTaken** [comment]
Does this tablet actually come with the plane? Or do the airlines install them? Because it seems like some are a bit different...

**Comment by merspieler** [comment]
Afaik, they are up to the airlines. They are also not really integrated into the systems like the OIS in the A350 is.

Last pic is interesting... (despite that it's an A330), it seems to be integrated into the table, similar to the Keyboard on the A380 and A350.

**Comment by AAL5665** [comment]
> Real:
> ![image](https://user-images.githubusercontent.com/10434911/114020521-97288280-9867-11eb-9594-9268218b8e31.png)
> ![image](https://user-images.githubusercontent.com/10434911/114020553-a0b1ea80-9867-11eb-860a-080159543215.png)
> ![image](https://user-images.githubusercontent.com/10434911/114020604-b0c9ca00-9867-11eb-9773-f08edb5faf24.png)
> ![image](https://user-images.githubusercontent.com/10434911/114020687-c2ab6d00-9867-11eb-936d-55c00019ea60.png)
> ![image](https://user-images.githubusercontent.com/10434911/114020768-d951c400-9867-11eb-9f6b-a429d8e0b971.png)
> ![image](https://user-images.githubusercontent.com/10434911/114020807-e1a9ff00-9867-11eb-96da-d427878ee0e9.png)
> ![image](https://user-images.githubusercontent.com/10434911/114020839-e8d10d00-9867-11eb-9886-ca9344c642f9.png)

Is this going to have to include Navigraph as well for the charts?

**Comment by merspieler** [comment]
Only if YOU pay for it

**Comment by Octal450** [comment]
Yep... ^

**Comment by NewoIsTaken** [comment]
I mean... we're not going to give you Navigraph for free...

**Comment by AAL5665** [comment]
I was asking whether it will require Navigraph for the charts.
@legoboyvdlp Regarding the first issue about the flightplan being displayed on the tablet:

![Charts](https://user-images.githubusercontent.com/74672200/114279096-68a2d700-9a22-11eb-8863-57bb3e69150f.png)

This is in the photo right?

**Comment by NewoIsTaken** [comment]
Ah, the US has charts for free published by the [FAA](https://www.faa.gov/air_traffic/flight_info/aeronav/digital_products/dtpp/search/), the UK has the [eAIP](https://www.aurora.nats.co.uk/htmlAIP/Publications/2021-03-25-AIRAC/html/index-en-GB.html), and I'm not sure for the rest and I am unsure if we do have the permissions to add them.

**Comment by merspieler** [comment]
[Australia](https://www.airservicesaustralia.com/aip/current/dap/AeroProcChartsTOC.htm)  
[Qatar](https://www.aim.gov.qa/eaip/2020-10-08-AIRAC/html/index-en-GB.html)
Have also some available.
